helix-core/src/path.rs

```diff
@@ -30,15 +30,7 @@ pub fn expand_tilde(path: &Path) -> PathBuf {
     path.to_path_buf()
 }
 
-/// Normalize a path, removing things like `.` and `..`.
-///
-/// CAUTION: This does not resolve symlinks (unlike
-/// [`std::fs::canonicalize`]). This may cause incorrect or surprising
-/// behavior at times. This should be used carefully. Unfortunately,
-/// [`std::fs::canonicalize`] can be hard to use correctly, since it can often
-/// fail, or on Windows returns annoying device paths. This is a problem Cargo
-/// needs to improve on.
-/// Copied from cargo: <https://github.com/rust-lang/cargo/blob/070e459c2d8b79c5b2ac5218064e7603329c92ae/crates/cargo-util/src/paths.rs#L81>
+/// Normalize a path (removing things like `.` and `..`) and resolving symlinks.
```

Comment:
I removed the `caution` because under the hood it still recognizes symlinks and uses `std::fs::canonicalize`.
https://github.com/helix-editor/helix/blob/2ceef99a8dfea19283a5bdb98d768a28136ef264/helix-core/src/path.rs#L40-L40